interview-questions.md

```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: 'this' is a designator used inside of an object or class in order to refer to the class or object one is inside of when calling a method or key in a key:value pair.
 
-  Researched answer:
+  Researched answer: I was sort of correct. It appears that 'this' refers to an object and when used inside of an object, it refers to the object itself. Outside of an object it either refers to the global object or is undefined.
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is a Javascript library which contains code necessary for building and combining components. Compenents are pieces of code in React which can be built to display HTML and run JSX commands on an HTML page. You can use React to build websites or apps which are both scalable and modular due to the various components used in their construction.
 
-  Researched answer:
+  Researched answer: React is not only a library, but a free and open-source front-end Javascript library. It is based on components and can be used to build single-page, mobile, or server-rendered apps.
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: Drawing a blank on this one.
 
-  Researched answer:
+  Researched answer: The constructor() method is required in a React class component because constructor handles state initialization and event handlers, which all affect how the component is rendered on the Document Object Model.
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX is a fusion of HTML and Javascript which React uses to render components to the DOM.
 
-  Researched answer:
+  Researched answer: Javascript XML (JSX) allows the programmer to write HTML in React. A notable difference between HTML and JSX is that HTML requires a method like getElementById() in order to incorporate Javascript by linking the HTML file to an external Javascript file. JSX on the other hand can incorporate both HTML and Javascript in one file, making rendering to the DOM quicker.
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Not totally sure on this one either.
 
-  Researched answer:
+  Researched answer: Yet Another Resource Negotiator (Yarn) is a package manager which can be used to automatically find, download, install, configure, upgrade, and uninstall a computer's packages. Yarn can be used to download and install libraries, plugins, etc. Yarn modifies yarn.lock and the README in a React application.
```

Comment:
Great job on this section.  Try to relate to something you can expand upon if you don't know the answer when possible.  For instance, you may not know what yarn is off the top of your head, but you likely can recall some yarn commands and when you use them.